interview-questions.md

```diff
@@ -8,63 +8,73 @@ Then, research the question to expand on your answer. Even if you feel you have
 1. What is a function? Why would you use one?
 
   Your answer:
-
+    It is a block of code that performs a specific task
   Researched answer:
-
+    The function is invoked only when you call the function.
 
 
 2. What is the difference between .map() and .filter()?
 
   Your answer:
+    .map() is a higher order function that returns a new array where you have applied a function on the element and changes the original element.  And .filter() also returns a new array of the elements but does not change the original values.
 
   Researched answer:
-
+    .filter() returns subset of the elements from the original array. and .map() produces and array with a new and different entries based on the elements in the original array.
 
 
 3. What is the difference between console.log() and return?
 
   Your answer:
+    console.log() strictly is for your terminal that logs whatever arguments you call.  and return is a statement that gives an output from a functions     
 
   Researched answer:
+    console.log() has no practical use to a program.  return allows you to pass the variable's value to the calling scope.
 
 
 
 4. In regards to functions, what is an argument?
 
   Your answer:
+    its an accessible object inside a function that has values passed into a function.
```

Comment:
Not exactly, an argument object and an object, are very specific things.  An object can be passed in AS an argument, but it is not specifically an argument as the interviewer is asking.  I think what you might be looking for is parameter.